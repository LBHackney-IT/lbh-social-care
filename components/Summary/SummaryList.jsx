import PropTypes from 'prop-types';

const SummaryList = ({ list }) => (
  <dl className="govuk-summary-list">
    {list &&
      list.map(({ key, title, value, href }) => (
        <div key={key} className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">{title}</dt>
          <dd className="govuk-summary-list__value">{value}</dd>
          {href && (
            <dd className="govuk-summary-list__actions">
              <a className="govuk-link" href="#">
                Change
                <span className="govuk-visually-hidden"> {title}</span>
              </a>
            </dd>
          )}
        </div>
      ))}
  </dl>
);

SummaryList.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      key: PropTypes.string.isRequired,
      title: PropTypes.string.isRequired,
      value: PropTypes.node.isRequired,
      href: PropTypes.string,
    }).isRequired
  ),
};

export default SummaryList;
