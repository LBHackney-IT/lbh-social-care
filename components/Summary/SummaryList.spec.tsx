import { render } from '@testing-library/react';

import SummaryList from './SummaryList';

describe('SummaryList component', () => {
  const props = {
    list: [
      {
        key: 'foo',
        title: 'Foo',
        value: 'f_o_o',
      },
      {
        key: 'bar',
        title: 'Bar',
        href: 'https://bar',
        value: 'f_o_o',
      },
      {
        key: 'foobar',
        title: 'FooBar',
        type: 'TextArea',
        value: 'asd\nasd\nasd\n',
      },
    ],
  };

  it('should render properly', () => {
    const { asFragment } = render(<SummaryList {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
