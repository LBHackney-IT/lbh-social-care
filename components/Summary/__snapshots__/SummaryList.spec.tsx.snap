// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummaryList component should render properly 1`] = `
<DocumentFragment>
  <dl
    class="govuk-summary-list"
  >
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Foo
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
        f_o_o
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        Bar
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
        f_o_o
      </dd>
      <dd
        class="govuk-summary-list__actions"
      >
        <a
          class="govuk-link"
          href="#"
        >
          Change
          <span
            class="govuk-visually-hidden"
          >
             Bar
          </span>
        </a>
      </dd>
    </div>
    <div
      class="govuk-summary-list__row"
    >
      <dt
        class="govuk-summary-list__key"
      >
        FooBar
      </dt>
      <dd
        class="govuk-summary-list__value"
      >
        <div
          class="outer"
        >
          <p
            data-testid="para"
          >
            asd
          </p>
          <p
            data-testid="para"
          >
            asd
          </p>
          <p
            data-testid="para"
          >
            asd
          </p>
        </div>
      </dd>
    </div>
  </dl>
</DocumentFragment>
`;
